Data quality issues
The first data quality issue I know exists is different spellings of the same DBA name. To attempt to counteract this I intend to change all DBA names to lowercase and remove any apostrophes. Noticed some DBA Names are in no way related to their aka name e.g. inpection id 2099050

Another issue I intend to fix is location data being stored as seperate values and also in a tuple 
From my initial inspections the location data in the tuple appears to be more accurate so removing the seperate location rows would appear to be the better option. 

I also noticed that some rows are missing information, the best option may be to ignore these rows as they are predominatly text based. If they were numerically based then you could potentially fill the missing values with a zero to keep the rows. 

In rows that are missing an AKA Name I will copy in the DBA Name. Also as the data is based on premises's in Chicago Illinois, any rows missing a state value will be populated with "IL". Similarly certain cases had a city value of "CHICAGOCHICAGO" so they were populated with the value "CHICAGO". Another issue with the city column is the inclusion of other cities. As the dataset is supposed to be based on the city of chicago this also needs to be rectified.

Premises with a result of Not ready and no entry and out of business columns have no data in the violations column
Some premises that Pass also have an empty violations column.

How to deal with rows with no location, latitude,longitude facility type , risk, INSPECTION TYPE

Another issue to be rectified is for all dates to be the same format. Some of the data in the date column is of type general in excel whereas some is of type date, so when ammending dates they must be in the format of dd/mm/yyyy. When a date that is in the form dd/mm/yy gets read in, it is automatically converted to dd/mm/yyyy during the reading of the data but when writing data back to the file it appears to get converted back the format of dd/mm/yy.

Using nltk remove stop words from names and aka names?
Compare name and aka name if theres a similarity remove the similarity and set both columns to the similarity

Removed all rows that had a city other than chicago
Then removed the city column completely
Removed the State column because they are all supposed to be facilities in chicago which is in illinois
Removed AKA Name because it is in the majority of cases the same as the DBA Name or a minor deviation from it


In [1]:
import csv
from collections import Counter
food = list(csv.DictReader(open('Food_Inspections.csv')))


In [3]:
dba_name = set()
for x in food:
    dba_name.add(x['DBA Name'])

print(len(dba_names))

NameError: name 'food' is not defined

In [48]:
new_food = []
for x in food:
    x['DBA Name'] = x['DBA Name'].lower()
    new_food.append(x)

In [49]:
dba_names = set()
for x in new_food:
    dba_names.add(x['DBA Name'])

In [50]:
dir(dba_names)
len(dba_names)

24610

Having converted all the DBA Names to lower case and creating a set of DBA Names, there are 24610 individual DBA Names

In [81]:
new_foods = []
i = 0
for x in new_food:
    x['DBA Name'] = x['DBA Name'].replace("'","")
    x['DBA Name'] = x['DBA Name'].replace(".","")
    x['DBA Name'] = x['DBA Name'].replace(",","")
    if x['AKA Name'] == '':
        x['AKA Name'] = x['DBA Name']
    elif x['City'] == '':
        x['City'] = 'CHICAGO'
    elif x['State'] == '':
        x['State'] = 'IL'
    new_foods.append(x)
    
dba = set()
for x in new_foods:
    dba.add(x['DBA Name'])


The number rows missing the state column was 168


In [35]:
{row['DBA Name'] for row in new_foods}

{'dereks dogs & deli llc',
 'target store t-3221',
 'la chusmita',
 'muller meats',
 'st malachy school',
 'caribou coffee company',
 'primo pizza inc',
 'nookies on wells inc',
 'yangos grill',
 'i monelli trattoria pizzeria',
 'cafe  el tapatio',
 'ryo sushi',
 'gateway foundation residential',
 'starrs fast food restaurant',
 'e-z convenience store inc',
 'vops llc',
 'daisy groceries',
 'west town pizza shop',
 'childrens house of miss tammie',
 'city dog',
 'leland elementary school',
 'kedzie food & meat',
 'carnitas la esquinita',
 'tastee freeze',
 'mr food mart inc',
 'halsted nursing center inc',
 'a-zmall world child development center',
 'la madrilena bakery',
 'indio cocina mexicana',
 'harczak sausages',
 'st anna bakery',
 'st elizabeth elementary school',
 '111 concession shop',
 'el tipico mexican grill',
 'marzeya bakery inc',
 'china ho restaurant',
 'jbj mart',
 'a & n food mart inc',
 'cafe espresso tc',
 'concord music hall',
 'darrins llc',
 'mangia roma',
 'tony

In [39]:
from collections import Counter
names = Counter(row['DBA Name'] for row in new_foods)

In [40]:
print(names)

Counter({'subway': 2313, 'dunkin donuts': 1196, 'mcdonalds': 977, '7-eleven': 430, 'chipotle mexican grill': 341, 'jimmy johns': 246, 'potbelly sandwich works llc': 243, 'au bon pain': 209, 'corner bakery cafe': 198, 'potbelly sandwich works': 197, 'mc donalds': 194, 'subway sandwiches': 189, 'dominos pizza': 183, 'dunkin donuts/baskin robbins': 181, 'sportservice soldier field': 176, 'whole foods market': 173, 'illinois sportservice inc': 167, 'harolds chicken shack': 165, 'mcdonalds restaurant': 164, 'freshii': 161, 'kfc': 148, 'see thru chinese kitchen': 144, 'sharks fish & chicken': 140, 'starbucks': 126, 'papa johns pizza': 122, 'pizza hut': 122, 'j & j fish': 118, 'popeyes chicken': 114, 'lou malnatis pizzeria': 114, 'levy restaurants at wrigley field': 111, 'dunkin donuts / baskin robbins': 106, 'citgo': 105, 'las islas marias': 102, 'popeyes': 101, 'protein bar': 98, 'kentucky fried chicken': 97, 'aramark': 97, 'host international inc': 95, 'argo tea': 95, 'taco burrito king': 

In [80]:
v =0
z = Counter()

for x in new_food:
    for y in x: 

        if x[y] == '':
            temp = x[y]
            z.update(temp)
            v += 1
for i in z:
    print(i)

In [2]:
s = Counter(row['City'] for row in food)
print(s)

Counter({'CHICAGO': 156126, 'Chicago': 258, '': 167, 'chicago': 79, 'CCHICAGO': 40, 'SCHAUMBURG': 20, 'MAYWOOD': 15, 'ELK GROVE VILLAGE': 12, 'CHicago': 10, 'CHICAGOCHICAGO': 8, 'EVANSTON': 8, 'CHESTNUT STREET': 8, 'SKOKIE': 8, 'INACTIVE': 8, 'CICERO': 6, 'CHARLES A HAYES': 6, 'CHCHICAGO': 6, 'ELMHURST': 5, 'WORTH': 5, 'OAK PARK': 4, 'CALUMET CITY': 4, 'SUMMIT': 4, 'NILES NILES': 3, 'SCHILLER PARK': 3, 'CHCICAGO': 3, 'CHICAGOI': 3, 'EAST HAZEL CREST': 3, 'ALSIP': 3, '312CHICAGO': 2, 'BEDFORD PARK': 2, 'BANNOCKBURNDEERFIELD': 2, 'CHICAGO HEIGHTS': 2, 'STREAMWOOD': 2, 'BERWYN': 2, 'BLUE ISLAND': 2, 'NAPERVILLE': 2, 'LAKE BLUFF': 1, 'BRIDEVIEW': 1, 'BLOOMINGDALE': 1, 'Norridge': 1, 'OOLYMPIA FIELDS': 1, 'WESTMONT': 1, 'JUSTICE': 1, 'TINLEY PARK': 1, 'LOMBARD': 1, 'COUNTRY CLUB HILLS': 1, 'BOLINGBROOK': 1, 'BURNHAM': 1, 'DES PLAINES': 1, 'LAKE ZURICH': 1, 'FRANKFORT': 1, 'OLYMPIA FIELDS': 1, 'alsip': 1, 'OAK LAWN': 1, 'GLENCOE': 1, 'BROADVIEW': 1, 'Maywood': 1, 'EVERGREEN PARK': 1})
